### PR TITLE
update to msal v2.0.1-preview

### PR DESCRIPTION
**Pull request by jennyf19**
update sample to msalv2.0.1-preview

Tested w/Android (system and embedded webview) and UWP (iOS simulator is not working...will try with a device soon)

Removed the `OnAppearing()` method because it was making an extra silent call every time the app appeared.

### CODE REVIEW

File: UserDetailsClient/UserDetailsClient/MainPage.xaml.cs [hunk 1]
```diff
@@ -14,47 +12,42 @@ namespace UserDetailsClient
...
+            AuthenticationResult authResult = null;
+            IEnumerable<IAccount> accounts = await App.PCA.GetAccountsAsync();
             try
             {
                 if (btnSignInSignOut.Text == "Sign in")
                 {
-                    AuthenticationResult ar = await App.PCA.AcquireTokenAsync(App.Scopes, App.UiParent);
-                    await RefreshUserDataAsync(ar.AccessToken);
-                    Device.BeginInvokeOnMainThread(() => { btnSignInSignOut.Text = "Sign out"; });
+                    // let's see if we have a user in our belly already
```
Comment (henrik-me) [inline UserDetailsClient/UserDetailsClient/MainPage.xaml.cs]: >belly [](start = 58, length = 5)

:)

File: UserDetailsClient/UserDetailsClient/MainPage.xaml.cs [hunk 2]
```diff
@@ -14,47 +12,42 @@ namespace UserDetailsClient
...
                 {
-                    IEnumerable<IAccount> accounts = await App.PCA.GetAccountsAsync();
-                    IAccount firstAccount = accounts.FirstOrDefault();
-                    await App.PCA.RemoveAsync(firstAccount);
-                  
+                    while (accounts.Any())
+                    {
+                        await App.PCA.RemoveAsync(accounts.FirstOrDefault());
+                        accounts = await App.PCA.GetAccountsAsync();
+                    }
```
Comment (henrik-me) [inline UserDetailsClient/UserDetailsClient/MainPage.xaml.cs]: This seems odd to me. Why sign-out of all accounts? Why not a specific/selected account? Aside: to me this means we should extend our API, RemoveAllAccounts / RemoveAccounts

File: UserDetailsClient/UserDetailsClient.iOS/packages.config [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="AsyncUsageAnalyzers" version="1.0.0-alpha003" targetFramework="xamarinios10" developmentDependency="true" />
-  <package id="Microsoft.Identity.Client" version="2.0.0-preview" targetFramework="xamarinios10" />
+  <package id="Microsoft.Identity.Client" version="2.0.1-preview" targetFramework="xamarinios10" />
+  <package id="Newtonsoft.Json" version="11.0.2" targetFramework="xamarinios10" />
```
Comment (henrik-me) [inline UserDetailsClient/UserDetailsClient.iOS/packages.config]: ><package id="Newtonsoft.Json" version="11.0.2" targetFramework="xamarinios10" /> [](start = 2, length = 80)

why?

File: UserDetailsClient/UserDetailsClient/MainPage.xaml.cs (hunk 1 above)
Comment (jennyf19) [reply UserDetailsClient/UserDetailsClient/MainPage.xaml.cs]: PM humor...I didn't add this comment.

File: UserDetailsClient/UserDetailsClient/MainPage.xaml.cs (hunk 2 above)
Comment (jennyf19) [reply UserDetailsClient/UserDetailsClient/MainPage.xaml.cs]: Good point. I'll sync w/JM about this when he gets back.

File: UserDetailsClient/UserDetailsClient.iOS/packages.config (hunk 3 above)
Comment (jennyf19) [reply UserDetailsClient/UserDetailsClient.iOS/packages.config]: was testing the iOS simulator w/this sample and needed to add in the Newtonsoft library, but should have removed it before pushing this change. thanks.

**Pull request merged**